### ISSUE TITLE
Segmentation fault when running optimization step with 3d convolution

### ISSUE THREAD

**Issue by cericson** [tensorflow/tensorflow#16139]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Arch Linux (4.14.13-1 linux kernel version)
- **TensorFlow installed from (source or binary)**: source (using the package here: https://www.archlinux.org/packages/community/x86_64/python-tensorflow-cuda/)
- **TensorFlow version (use command below)**: 1.4.1
- **Python version**: 3.6.4
- **Bazel version (if compiling from source)**: 0.9.0
- **GCC/Compiler version (if compiling from source)**: 7.2.1
- **CUDA/cuDNN version**: 9.1.85-1/7.0.5-2
- **GPU model and memory**: NVidia Quadro K4200, 4028MiB
- **Exact command to reproduce**: `python test.py`
Note that the same code also fails in a Ubuntu docker container (Dockerfile attached).

### Describe the problem
I set up a computation graph with a 3d convolution. I can evaluate the result of this graph, but when I attempt to optimize the parameters of the graph (`train_step.run(feed_dict={x: sample, y_: label})`), tensorflow segfaults.

In a jupyterlab notebook running on Ubuntu, if I run the same code, the notebook hangs indefinitely at the same line. In both cases, the last line of the program is never run - "ran train step" is never printed.

I also tried running this on my CPU with `os.environ['CUDA_VISIBLE_DEVICES'] = '-1'`. I get the same segfault.

The segfault goes away if I do any of the following:
- Remove the 3d convolution
- Reduce the input size significantly (e.g. 100x smaller to 1 x 41 x 96 x 128 x 1)
- Reduce the kernel size significantly

### Source code / logs
Minimal example code (test.py):
```python
import numpy as np
import tensorflow as tf

sample = np.zeros((1, 41, 960, 1280, 1))
label = np.zeros((1,))

rc_kernel = np.ones((31,))

x = tf.placeholder(tf.float64, shape=[None, 41, 960, 1280, 1])
y_ = tf.placeholder(tf.float64, shape=[None])

W_conv_r = tf.Variable(rc_kernel.reshape((1, -1, 1, 1, 1)))
h_blur = tf.nn.conv3d(x, W_conv_r, [1, 1, 1, 1, 1], "VALID")

h_sum = tf.reduce_sum(tf.reduce_sum(tf.reduce_sum(h_blur, axis=3), axis=2), axis=1)
y = tf.sigmoid(h_sum)

sq_err = (y - y_) ** 2

train_step = tf.train.GradientDescentOptimizer(0.1).minimize(sq_err)

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    E = sq_err.eval(feed_dict={x: sample, y_: label})
    print(f'E = {E}')
    train_step.run(feed_dict={x: sample, y_: label})  # fails here
    print('ran train step')
```

Dockerfile:
[Dockerfile.txt](https://github.com/tensorflow/tensorflow/files/1633128/Dockerfile.txt)

**Comment by aselle** [comment]
@benoitsteiner, any other thoughts... here is a sampling of the eigen function stack entries that are so prevelant

```
#30817 0x00007fffd55898fb in Eigen::TensorEvaluator<Eigen::TensorContractionOp<Eigen::array<Eigen::IndexPair<long>, 1ul> const, Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>::Context<Eigen::internal::gemm_pack_lhs<double, long, Eigen::internal::TensorContractionSubMapper<double, long, 1, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, 4, 2, 0, false, false>, Eigen::internal::gemm_pack_rhs<double, long, Eigen::internal::TensorContractionSubMapper<double, long, 0, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, 4, 0, false, false>, Eigen::internal::gebp_kernel<double, double, long, Eigen::internal::blas_data_mapper<double, long, 0, 0>, 4, 4, false, false>, Eigen::internal::TensorContractionInputMapper<double, long, 1, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, Eigen::internal::TensorContractionInputMapper<double, long, 0, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, Eigen::internal::blas_data_mapper<double, long, 0, 0> >::pack_lhs(long, long) ()
   from /usr/local/google/home/aselle/tensorflow3/local/lib/python2.7/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so
#30818 0x00007fffd5588a49 in Eigen::TensorEvaluator<Eigen::TensorContractionOp<Eigen::array<Eigen::IndexPair<long>, 1ul> const, Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>::Context<Eigen::internal::gemm_pack_lhs<double, long, Eigen::internal::TensorContractionSubMapper<double, long, 1, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, 4, 2, 0, false, false>, Eigen::internal::gemm_pack_rhs<double, long, Eigen::internal::TensorContractionSubMapper<double, long, 0, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, 4, 0, false, false>, Eigen::internal::gebp_kernel<double, double, long, Eigen::internal::blas_data_mapper<double, long, 0, 0>, 4, 4, false, false>, Eigen::internal::TensorContractionInputMapper<double, long, 1, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, Eigen::internal::TensorContractionInputMapper<double, long, 0, Eigen::TensorEvaluator<Eigen::TensorReshapingOp<Eigen::DSizes<long, 2> const, Eigen::TensorVolumePatchOp<-1l, -1l, -1l, Eigen::TensorMap<Eigen::Tensor<double const, 5, 1, long>, 16, Eigen::MakePointer> const> const> const, Eigen::ThreadPoolDevice>, Eigen::array<long, 1ul>, Eigen::array<long, 1ul>, 2, true, false, 0, Eigen::MakePointer>, Eigen::internal::blas_data_mapper<double, long, 0, 0> >::enqueue_packing_helper(long, long, long, bool) ()
   from /usr/local/google/home/aselle/tensorflow3/local/lib/python2.7/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so

```

**Comment by amathuri** [comment]
Could this bug be related to the following issue?
https://github.com/tensorflow/tensorflow/issues/14807

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by cericson** [comment]
@amathuri looks related to me. I must have missed that when searching for a solution to this problem.

If it's helpful to anyone else, I did end up coming up with a workaround for my particular case. My architecture essentially has a separable 3D convolution, so I only need to convolve in one dimension at a time. My workaround was to use tf.nn.conv2d() instead, reshaping the z dimension into the batch dimension beforehand (so the shape is (depth, height, width, channels)). This lets me convolve in the x and y dimensions. Convolving in the z dimension is a little trickier, but I did it by swapping the x and z dimensions with tf.transpose(), convolving in x, then swapping the dimensions back.

Unfortunately, this method prevents batching, so won't be helpful for #14807.

**Comment by reedwm** [comment]
Pinging @benoitsteiner

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 99 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 115 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 130 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 145 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 160 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 175 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 175 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 190 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 20 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @poxvoculi, @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
This issue is replicating  in [2.8](https://colab.sandbox.google.com/gist/mohantym/f7b0f5294fcbf44bc0632ebddb897eb7/github_16139.ipynb#scrollTo=8cIAhnfez68P) version too.

**Comment by mihaimaruseac** [comment]
@mohantym did you reproduce using TF v2 APIs?

**Comment by mohantym** [comment]
Hi @mihaimaruseac ! I replicated using the migration document in compat.v1 mode . have not tried v2 APIs yet.

**Comment by mihaimaruseac** [comment]
We no longer support v1 APIs.

**Comment by mohantym** [comment]
Hi @cericson ! 1.x versions are not supported any more . Can you please create a new issue with the 2.x code base?

**Comment by cericson** [comment]
So I refactored the original code a bit (see below) to run it in tensorflow 2.8.0 with no issue. The bug may have been resolved already, or at least it doesn't resurface with the way I'm using the v2 API.
```
import tensorflow as tf
import numpy as np

sample = np.zeros((1, 41, 960, 1280, 1))
label = np.zeros((1,))
kernel = tf.Variable(np.ones((1, 31, 1, 1, 1)))

with tf.GradientTape() as g:
    h_blur = tf.nn.conv3d(sample, kernel, [1, 1, 1, 1, 1], "VALID")
    h_sum = tf.reduce_sum(tf.reduce_sum(tf.reduce_sum(h_blur, axis=3), axis=2), axis=1)
    y = tf.sigmoid(h_sum)
    sq_err = (y - label) ** 2

grad = g.gradient(sq_err, kernel)
optimizer = tf.optimizers.SGD()
optimizer.apply_gradients([(grad, kernel)])
print('ran train step')
```

**Comment by mohantym** [comment]
@cericson ! This issue is getting resolved with input shape (1, 41, 96, 128, 1)  or (1, 41, 400, 800, 1) as you pointed out earlier but it is  replicating with a shape (1, 41, 960, 1280, 1). Attaching [gist](https://colab.sandbox.google.com/gist/mohantym/f2201df9a33c41ac62ddf3bca8be4549/github_16139_2-8.ipynb#scrollTo=Fq_5wbxlch_m) for reference.
